event.c

```diff
@@ -676,17 +676,17 @@ static void event_image_preview(struct context *cnt,
 
         if ((cnt->ffmpeg_output || (cnt->conf.movie_extpipe_use && cnt->extpipe)) && !use_imagepath) {
             if (cnt->conf.movie_extpipe_use && cnt->extpipe) {
-                basename_len = strlen(cnt->extpipefilename) + 1;
-                strncpy(previewname, cnt->extpipefilename, basename_len);
+                basename_len = MIN(strlen(cnt->extpipefilename) + 1, sizeof(previewname) - 1);
+                strncpy(previewname, cnt->extpipefilename, sizeof(previewname));
                 previewname[basename_len - 1] = '.';
             } else {
                 /* Replace avi/mpg with jpg/ppm and keep the rest of the filename. */
-                basename_len = strlen(cnt->newfilename) - 3;
-                strncpy(previewname, cnt->newfilename, basename_len);
+                basename_len = MIN((cnt->newfilename - strrchr(cnt->newfilename, '.')) + 1, sizeof(previewname) - 1);
+                strncpy(previewname, cnt->newfilename, sizeof(previewname));
```

Comment:
Same situation as before:

    snprintf(previewname, sizeof(previewname), "%s%s", cnt->newfilename, imageext(cnt));

And as before, the `basename_len` line can be removed as unnecessary.